homeassistant/components/stream/core.py

```diff
@@ -110,41 +120,71 @@ def idle(self) -> bool:
         return self._idle_timer.idle
 
     @property
-    def segments(self) -> list[int]:
+    def sequences(self) -> list[int]:
         """Return current sequence from segments."""
         return [s.sequence for s in self._segments]
 
+    @property
+    def last_segment(self) -> Segment:
+        """Return the last segment without iterating."""
+        if self._segments:
+            return self._segments[-1]
+        return None
+
+    @property
+    def last_sequence(self) -> int:
+        """Return the last sequence number without iterating."""
+        if self._segments:
+            return self._segments[-1].sequence
+        return -1
+
     @property
     def target_duration(self) -> int:
         """Return the max duration of any given segment in seconds."""
-        segment_length = len(self._segments)
-        if not segment_length:
-            return 1
-        durations = [s.duration for s in self._segments]
-        return round(max(durations)) or 1
+        durations = [s.duration for s in self._segments if s.duration]
```

Comment:
Needed to add the `if s.duration` as the last incomplete segment will have a duration of 0 until it is finalized.
Also changed this to return `MIN_SEGMENT_DURATION` instead of just 1.

Reply:
I added a s.complete property which I used here and a few other places in order to make things more clear.